### PR TITLE
Airlock Electronic Installation Speed Change

### PR DESCRIPTION
**Pull request by DorainGrey**
## What this does
Crowbaring, Inserting and screwing airlock electronics are now instantaneous.

## Why it's good
Repairing Emagged, or otherwise broken airlocks is painfully slow, this poses a problem when you have a mass emagger, ninja, grue etc and usually means doors get left unfixed. 
There is no way to speed up this task, unlike welding which is by default 2 seconds in most scenarios and can be speed up by using a better tool. 
These changes address this issue by making the overall process faster and more in line with other engineering tasks.

## Changelog
:cl:
 * tweak: Crowbaring, Inserting and screwing airlock electronics are now instantaneous.

### CODE REVIEW

**Comment by Kurfursten** [comment]
You speak of better tools, why not make an upgraded solder that works faster?

4 seconds is consistent with all other repair actions. Why is soldering exceptional compared to welding, wrenching, etc? And why 2 seconds? Why not instant if mass emagging is such an impossibly high importance task, even more than rebuilding walls in a breach (4sec).

I also disagree with your general propositions that most welding is 2 seconds and also that there's no way to speed up soldering. You can solder multiple boards at once, which I recommend you do.

**Comment by DorainGrey** [comment]
> You speak of better tools, why not make an upgraded solder that works faster?

I'd need to sprite it, re-write the function and then map it in. It's a relatively niche mechanic and I really don't see any reason it should be the slow by default regardless. 

> 4 seconds is consistent with all other repair actions. Why is soldering exceptional compared to welding, wrenching, etc? And why 2 seconds? Why not instant if mass emagging is such an impossibly high importance task, even more than rebuilding walls in a breach (4sec).

For a start welding can be sped up, soldering cannot as of now, and plenty of welding and general repair actions are 2 seconds or instant.
I don't really see the relevance of breaches either, why would you seal a breach with a reinforced wall, not an RCD normal wall(2 sec), Inflatables(Instant), atmos tape(instant) or any of the other methods. What does breaches have to do with doors at all?
Breaking or dissembling a reinforced wall is not comparable  to an airlock which can and does get emagged in an instant or broken by antags in large quantities.
I find it very hard to justify not just de-constructing door with RCD and constructing a new, since that tends to be faster.

As for the specific 2 seconds, it felt appropriate so that there's some time-sink into the process. It's still a 4-5 step process and is not quick by any stretch of imagination.

**Comment by Kurfursten** [comment]
> I'd need to sprite it, re-write the function and then map it in. It's a relatively niche mechanic and I really don't see any reason it should be the slow by default regardless.

Why would you map it in? Make upgraded soldering tools an R&D item. Is spriting really holding you up? Just bucket tool the red to purple or something. I'll even do it for you, ten seconds tops.

> For a start welding can be sped up, soldering cannot as of now

Wrong, see last post; moreover, this doesn't make any sense. Why would you say, "We shouldn't make a faster version of the solder, because there isn't a faster alternative to a solder". That's *the only* situation you want to write a faster alternative.

> I don't really see the relevance of breaches either... What does breaches have to do with doors at all?

What? We're comparing "common things antags destroy en masse", which includes walls. Building and repairing walls takes 4 seconds (actually 8, 4 for the girder and 4 for the plating). But 4 for each action.

> Breaking or dissembling a reinforced wall is not comparable to an airlock which can and does get emagged in an instant or broken by antags in large quantities.

Meteors and bombs are actually much more common than a mass emag incident. I would know - I wrote the soldering tool. The reason this is the only server that has such a tool is because of me. You're right that mass emagging is extremely niche. 4 seconds is fine in 99.98% of rounds, but I could see the appeal of having a special tool at R&D for the one in a couple hundred where someone goes wild.

> Why not just deconstruct

That's your valid alternative. You make a choice - do I want to just RCD down a door with no access, or solder it, do it the right way. Aren't you just now disproving your own statement when you said there was no fast alternative to soldering? You give "inflatables and tape" as valid alternatives to constructing a wall, but RCDing a door doesn't count?

**Comment by WindowsErrors** [comment]
i just want the soldering iron to not take as long

**Comment by pedr69** [comment]
or just have the final step, screwdrivering the thing closed be instant, so it's like all the other screwdrivering

**Comment by xeroxemnas** [comment]
solder gang solder gang

**Comment by DorainGrey** [comment]
> Why would you map it in? Make upgraded soldering tools an R&D item. Is spriting really holding you up? Just bucket tool the red to purple or something. I'll even do it for you, ten seconds tops.

Well the improved welding tools are mapped in, locking things behind RD given the server culture tends to mean they are rarely used. If it was instant sure I could see locking it behind RD being balanced but for a small benefit I don't see why it shouldn't be available. I won't rule it out, just doesn't seem necessary?

> Wrong, see last post; moreover, this doesn't make any sense. Why would you say, "We shouldn't make a faster version of the solder, because there isn't a faster alternative to a solder". That's _the only_ situation you want to write a faster alternative

What do you mean wrong, welding can be sped up. We have improved welding tools, which speed the process up. 


> What? We're comparing "common things antags destroy en masse", which includes walls. Building and repairing walls takes 4 seconds (actually 8, 4 for the girder and 4 for the plating). But 4 for each action.

Yes but crucially, if a segment of wall is broken. You can easily multitask and do 3 segments more or less at once and fix other stuff at the same time. This is especially true with all the fun stuff like clarkes and the like. There is nothing to speedup fixing an emag, you can't even change hands during parts of it.


> Meteors and bombs are actually much more common than a mass emag incident. I would know - I wrote the soldering tool. The reason this is the only server that has such a tool is because of me. You're right that mass emagging is extremely niche. 4 seconds is fine in 99.98% of rounds, but I could see the appeal of having a special tool at R&D for the one in a couple hundred where someone goes wild.

You say this is uncommon but I haven't even played much since this PR and I've had to deal with 3 mass emagging incidents, even taking a quick look at the logs shows it's common enough. Sure welder bombs are more common but they are easier to fix and aren't exactly stealthy assuming someone even survives. I fixed 20+ doors today it was hell.  
Not that I think it being uncommon is an excuse to not improve something either

> That's your valid alternative. You make a choice - do I want to just RCD down a door with no access, or solder it, do it the right way. Aren't you just now disproving your own statement when you said there was no fast alternative to soldering? You give "inflatables and tape" as valid alternatives to constructing a wall, but RCDing a door doesn't count?

RCDing is one way of doing it but surely we'd like to encourage players to you know actually fix stuff... 
Instead of just going down the TG route of RCD's being the best for everything.

**Comment by DorainGrey** [comment]
I think I will follow through on pedr's suggestion tomorrow however and instead restore soldering speed but make the insertion of airlock electronics and screwdriver instant.
It fits better.
Will that address your objections?

**Comment by Kurfursten** [comment]
Yes that's fine with me, and it also leaves the door open for an improved solder tool in the future which I think would be a fun bonus starting tool for the Chief Engineer + make at R&D/mechanics

I really feel that the mechanic could use more upgraded engineering tools to outfit the engineering staff and strengthen his role as a support engineer

**Comment by DorainGrey** [comment]
This has been updated @Kurfursten 

- Soldering speed has been restored to original.
- Crowbaring, inserting and screwing airlock electronics have been made instantaneous.
- Wiring and wrenching airlocks have been left alone, this means it takes a decent bit longer to fix an esword or alien destroyed airlock, this should help maintain balance and not cuck xeno's or ninja's as breaking airlocks is a key part of their gameplay.

I'm glad we did it this way now, it helps maintain a balance of making mass emagging much easier to deal with but not impacting other airlock damage too much.

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -1249,11 +1249,9 @@ About the new airlock wires panel:
...
 			user.visible_message("[user] removes the electronics from the airlock assembly.", "You start to remove electronics from the airlock assembly.")
 			// TODO: refactor the called proc
-			if (do_after(user, src, 20))
-				to_chat(user, "<span class='notice'>You removed the airlock electronics!</span>")
-				revert(user,null)
-				qdel(src)
-				return
+			to_chat(user, "<span class='notice'>You removed the airlock electronics!</span>")
+			revert(user,null)
+			qdel(src)
```
Comment (Kurfursten) [inline code/game/machinery/doors/airlock.dm:1254]: no more early return?

Comment (DorainGrey) [reply code/game/machinery/doors/airlock.dm:1254]: Missed that, fixed. Adding fingerprints when installing electronics probably isn't wanted.

**Pull request merged**